kernel/sched.c

```diff
@@ -1621,8 +1621,8 @@ static inline int z_vrfy_k_is_preempt_thread(void)
 
 #ifdef CONFIG_SCHED_CPU_MASK
 # ifdef CONFIG_SMP
-/* Right now we use a single byte for this mask */
-BUILD_ASSERT(CONFIG_MP_MAX_NUM_CPUS <= 8, "Too many CPUs for mask word");
+/* Right now we use a single DWORD for this mask */
+BUILD_ASSERT(CONFIG_MP_MAX_NUM_CPUS <= 32, "Too many CPUs for mask word");
```

Comment:
I wonder do we need this assert at all? It is already handled through Kconfig, right?

Reply:
+1 to removal, this is no doubt descended from some early code without a limit in kconfig.

Reply:
Build_ASSERT is checked in compile time, right?
Is there any drawback to keep it?

Reply:
> Build_ASSERT is checked in compile time, right? Is there any drawback to keep it?

Unnecessary code & redundancy. It also gives the false impression that the bounds are not enforced anywhere else even though they are (in Kconfig).